oidc-authentication.md

```diff
@@ -63,6 +63,183 @@
+
+##### authentication flow
+[OpenID Connect] built on top of OAuth 2's Authorization Grants, and
+introduced the concept of Authentication Flow - a process where a [client] can
+use a valid grant (such as the Implicit Grant, Authorization Code Grant, or
+others) to perform [authentication].
+
+The end result of any successful authentication flow is that the client
+performing it receives one or more credentials from the [identity provider]. In
+the case of classic OpenID Connect flows, those credentials are typically either
+the [id token], an [access token] specific to a particular resource server, or
+both.
+
+In addition, OpenID Connect provides a way for a [relying party] to request
+arbitrary [claims] or [credentials] inside the ID Token. In particular, this
+specification uses that mechanism to allow a  [relying party] to request one or
+more reusable credentials that enable audience-constrained [protected resource
+requests] to multiple unaffiliated [resource servers] (see section [Multi
+Resource Server (Multi-RS) Use Case] for more details).
+
+##### Web ID
+A globally unique [identifier] (typically an HTTPS URL) for an [entity], that
+can be resolved to a Web ID Profile document.
+
+##### user-agent
+As defined by RFC 2616 [RFC2616]. Often, a formal name for a web browser. Note
+that this is often separate from a [client] application (in many cases, client
+apps written in Javascript run inside the browser).
+
+##### resource controller
+An [entity] with administrative permissions to a resource (e.g. `acl:Control`
+access mode in the case of [WAC]). Alternatively, the [entity] in control of the
+[resource server] (such as a [Solid pod] or storage space) in which the resource
+resides.
+
+Similar to the OAuth 2 and
+[UMA 2](https://docs.kantarainitiative.org/uma/wg/rec-oauth-uma-grant-2.0.html)
+concept of "resource owner", the resource controller MAY be an [end-user]
+(natural person) or MAY be a non-human entity treated as a person for limited
+legal purposes (legal person), such as a corporation.
+
+Note the use of "controller" as opposed to "owner"; this specification (and
+Solid in general), has no notion of owner.
+
+A given resource can have zero, one, or multiple resource controllers.
+
+##### protected resource
+A resource to which access is constrained by an appropriate authorization
+mechanism (such as Web Access Control [WAC] rules or Authorization Capabilities
+[zCAP]).
+
+##### client
+An application, service, bot, or other software entity interacting with a
+protected resource hosted by a [resource server] on behalf of a [requesting
+party]. The term "client" does not imply any particular implementation
+characteristics (such as whether the application executes on a server, a
+desktop, inside a web browser, or on other devices).
+
+The client is an [entity] in its own right, possessing an [identifier], and in
+some cases, a corresponding [controller document] containing client metadata.
```

Comment:
I find this use of *controller document* confusing. I think we  should clearly separate software entities from social entities. Preferably *controller* would get used only for social entities. Since we consider having document for both, how about just calling it *entity document* or *description document*?

Reply:
Some prior conversation in https://github.com/solid/specification/issues/16 . Something one may call in non-normative way *solid pod* could also provide way to deploy apps and bots. I think we should leave that term non-normative and not constrained to storage + OP composition.

Reply:
While I can see the reason for being a bit more specific about the document naming, I think `entity document` and `description document` would be poor descriptions.

 - As described both software and social entities can have "controller documents." I'd say the term "controller document" should stay around to just describe the document that comes from a WebID or DID regardless of what it represents.
 - Entity document is a poor description because both the user and the client are entities
 - Description document is a poor description because in both cases they provide more than a description, they also represent the entity.
 - We could just call it the `end-user controller document` and the `client controller document`

Reply:
I mostly worry about using *controller* in both
* resource controller
* controller document

Would we ever see case for having a client application acting as *resource controller*? The *resource controller* definition says:
> concept of "resource owner", the resource controller MAY be an [end-user]
(natural person) or MAY be a non-human entity treated as a person for limited
legal purposes (legal person), such as a corporation.

Which suggests that we would expect some *social entity* here not *software entity*.

Reply:
I agree with this sentiment. It makes sense to keep naming "resource controller" as such. However "controller document" should be changed to avoid confusion.

Reply:
Based on the call, we think that "entity document" is a better name

Reply:
Based on further discussions, the current proposed term (in this PR) is "identifier document". 
This is to match the existing terms "Web ID (Identifier) Document" and "DID (Decentralized Identifier) Document".